### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/pagmo2/commit/f6895e547c808e60e46f3db186d8400217ac0ac7#commitcomment-77051465

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/pagmo2/commit/49c42d01a748ce7790d4676f6e4e0e8182cdf66e

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

I will be annotating this PR, and I fully expect to be asked to drop certain files.

### CODE REVIEW

File: doc/doxygen/Doxyfile.in [hunk 1]
```diff
@@ -753,7 +753,7 @@ SHOW_NAMESPACES        = YES
 # The FILE_VERSION_FILTER tag can be used to specify a program or script that
 # doxygen should invoke to get the current version for each file (typically from
 # the version control system). Doxygen will invoke the program by executing (via
-# popen()) the command command input-file, where command is the value of the
+# popen()) the command input-file, where command is the value of the
```
Comment (jsoref) [inline doc/doxygen/Doxyfile.in:756]: I'm not particularly certain about this change, and it's possible the file is from upstream doxygen and shouldn't be touched.

File: doc/sphinx/docs/cpp/utils/genetic_operators.rst [hunk 2]
```diff
@@ -24,7 +24,7 @@ Some utilities to perform common genetic operations
    :exception std\:\:invalid_argument: if *bounds* contain any nan or infs.
    :exception std\:\:invalid_argument: if any lower bound is greater than the corresponding upper bound.
    :exception std\:\:invalid_argument: if *nix* is larger than bounds size.
-   :exception std\:\:invalid_argument: if any of the integer bounds are not not integers.
+   :exception std\:\:invalid_argument: if any of the integer bounds are not integers.
```
Comment (jsoref) [inline doc/sphinx/docs/cpp/utils/genetic_operators.rst:27]: double negatives are confusing.

File: doc/sphinx/docs/cpp/utils/multi_objective.rst [hunk 3]
```diff
@@ -4,7 +4,7 @@ Multi-objective optimization utilities
 ======================================
 
 A number of utilities to compute quantities that are of relevance to
-the determination of non dominated fronts, Pareto dominance criterias and
+the determination of non dominated fronts, Pareto dominance criteria and
```
Comment (jsoref) [inline doc/sphinx/docs/cpp/utils/multi_objective.rst:7]: the singular is criterion

File: doc/sphinx/overview.rst [hunk 4]
```diff
@@ -109,7 +109,7 @@ Preconditioned truncated Newton (from NLopt)           :cpp:class:`pagmo::nlopt`
 Shifted limited-memory variable-metric (from NLopt)    :cpp:class:`pagmo::nlopt`                                                                     S-U
 Ipopt                                                  :cpp:class:`pagmo::ipopt`                                                                     S-CU
 SNOPT (in pagmo_plugins_non_free affiliated package)   `pagmo::snopt7 <https://esa.github.io/pagmo_plugins_nonfree/cpp_snopt7.html>`__               S-CU
-WORHP (in pagmo_plugins_non_free affiliated package)   `pagmo::wohrp <https://esa.github.io/pagmo_plugins_nonfree/cpp_worhp.html>`__                 S-CU
+WORHP (in pagmo_plugins_non_free affiliated package)   `pagmo::worhp <https://esa.github.io/pagmo_plugins_nonfree/cpp_worhp.html>`__                 S-CU
```
Comment (jsoref) [inline doc/sphinx/overview.rst:112]: This seemed odd... it doesn't agree w/ the two other instances on the same line, nor is it mentioned in the linked document.

File: include/pagmo/algorithms/cstrs_self_adaptive.hpp [hunk 5]
```diff
@@ -157,7 +157,7 @@ PAGMO_DLL_PUBLIC std::ostream &operator<<(std::ostream &, const penalized_udp &)
  * approach is an alternative to using the meta-problem pagmo::unconstrain to transform the
  * constrained fitness into an unconstrained fitness.
  *
- * The self-adaptive constraints handling meta-algorithm is largely based on the ideas of Faramani and Wright but it
+ * The self-adaptive constraints handling meta-algorithm is largely based on the ideas of Farmani and Wright but it
```
Comment (jsoref) [inline include/pagmo/algorithms/cstrs_self_adaptive.hpp:160]: https://www.researchgate.net/publication/3418762_Self-Adaptive_Fitness_Formulation_for_Constrained_Optimization

File: include/pagmo/algorithms/ipopt.hpp [hunk 6]
```diff
@@ -83,7 +83,7 @@ PAGMO_DLL_PUBLIC unsigned ipopt_internal_test();
  * set_selection(population::size_type), set_replacement(const std::string &) and
  * set_replacement(population::size_type).
  *
- * Configuring the optimsation run
+ * Configuring the optimization run
  * -------------------------------
```
Comment (jsoref) [inline include/pagmo/algorithms/ipopt.hpp:87]: I didn't adjust the `-`s. I can ....

File: include/pagmo/problems/hock_schittkowski_71.hpp [hunk 7]
```diff
@@ -59,7 +59,7 @@ namespace pagmo
  * Lecture Notes in Economics and Mathematical Systems, 187, 1981. doi: 10.1007/978-3-642-48320-2.
  *
  */
-struct PAGMO_DLL_PUBLIC hock_schittkowsky_71 {
+struct PAGMO_DLL_PUBLIC hock_schittkowski_71 {
```
Comment (jsoref) [inline include/pagmo/problems/hock_schittkowski_71.hpp:62]: The reference on line 58 is to:
> * See: W. Hock and K. Schittkowski. Test examples for nonlinear programming codes.

https://link.springer.com/book/10.1007/978-3-642-48320-2

I couldn't see any explanation of why to reference w/ a `-y`.

Google has this to say when I try:
> Showing results for [Hock Schittkowski](https://www.google.com/search?sxsrf=ALiCzsYPY852I44O_A7g_uIwqjjcpM_Drg:1656305218904&q=Hock+Schittkowski&spell=1&sa=X&ved=2ahUKEwiMkNnh6cz4AhURb80KHfCJDB8QkeECKAB6BAgBEDM)
> Search instead for [Hock Schittkowsky](https://www.google.com/search?sxsrf=ALiCzsYPY852I44O_A7g_uIwqjjcpM_Drg:1656305218904&q=Hock+Schittkowsky&nfpr=1&sa=X&ved=2ahUKEwiMkNnh6cz4AhURb80KHfCJDB8QvgUoAXoECAEQNA)

File: CMakeLists.txt [hunk 8]
```diff
@@ -194,7 +194,7 @@ set(PAGMO_SRC_FILES
     "${CMAKE_CURRENT_SOURCE_DIR}/src/problems/cec2009.cpp"
     "${CMAKE_CURRENT_SOURCE_DIR}/src/problems/schwefel.cpp"
     "${CMAKE_CURRENT_SOURCE_DIR}/src/problems/rosenbrock.cpp"
-    "${CMAKE_CURRENT_SOURCE_DIR}/src/problems/hock_schittkowsky_71.cpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/problems/hock_schittkowski_71.cpp"
```
Comment (jsoref) [inline CMakeLists.txt:197]: The files are renamed. 

See my comment on:
> include/pagmo/problems/hock_schittkowski_71.hpp

File: include/pagmo/problems/luksan_vlcek1.hpp [hunk 9]
```diff
@@ -26,8 +26,8 @@ You should have received copies of the GNU General Public License and the
 GNU Lesser General Public License along with the PaGMO library.  If not,
 see https://www.gnu.org/licenses/. */
 
-#ifndef PAGMO_PROBLEMS_LUKSAN_VLCECK1_HPP
-#define PAGMO_PROBLEMS_LUKSAN_VLCECK1_HPP
+#ifndef PAGMO_PROBLEMS_LUKSAN_VLCEK1_HPP
+#define PAGMO_PROBLEMS_LUKSAN_VLCEK1_HPP
```
Comment (jsoref) [inline include/pagmo/problems/luksan_vlcek1.hpp:30]: This matches the filename...

**Review by jsoref — COMMENTED** [review]
Most changes were automatically suggested by Google Sheets (including the names... It knows a lot more about domains than I do...). But all fault is mine. And I ran into some tooling glitches that I don't understand, so it wouldn't shock me if something went wrong.

I think I made a mental note to ask about one other item, but it isn't jumping out at me atm...

**Comment by bluescarni** [comment]
@jsoref thank you, that looks great! The build failure in the CI is due to our Doxyfile needing an update, I'll update the PR later tonight and trigger the rebuild.

**Comment by jsoref** [comment]
Hi @bluescarni, just checking in. (I'm in no hurry. I just accidentally reopened this window...)

**Comment by bluescarni** [comment]
@jsoref sorry for taking so long... thanks again for taking care of all this.

**Pull request merged**